### PR TITLE
Use typscript 2 features for type definitions in react starter

### PR DESCRIPTION
**Pull request by antoinegslight**
Easier managing of typescript declaration files with typescript 2.0+ and update almost every dependencies in the react-starter-pack.

See [](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md) for informations on getting declarion files with npm.

### CODE REVIEW

File: react-starter/package.json [hunk 1]
```diff
@@ -3,23 +3,21 @@
...
   "description": "",
   "main": "index.js",
-  "scripts": {
-    "postinstall": "typings install"
-  },
   "author": "wfortin & vseguin",
   "license": "MIT",
   "devDependencies": {
+    "@types/react": "^15.0.1",
+    "@types/react-dom": "^0.14.20",
```
Comment (antoinegslight) [inline react-starter/package.json:10]: These replace the declarations files stated in typings.json

File: react-starter/package.json [hunk 2]
```diff
@@ -3,23 +3,21 @@
...
     "gulp": "^3.9.1",
     "gulp-rename": "^1.2.2",
-    "gulp-sass": "^2.3.1",
-    "ts-loader": "^0.8.2",
-    "typescript": "^1.8.10",
-    "typings": "^1.0.4",
```
Comment (antoinegslight) [inline react-starter/package.json:17]: Typings is no longer necessary

File: react-starter/tsconfig.json [hunk 3]
```diff
@@ -7,9 +7,6 @@
     "jsx": "react",
     "outDir": "dist"
   },
-  "files": [
-    "./typings/index.d.ts"
```
Comment (antoinegslight) [inline react-starter/tsconfig.json:11]: This file will no longer necessary, it will not be created...

**Comment by wfortin** [comment]
Thanks, closing this as it was applied in #3

**Pull request closed without merging**